hub/api/tests/test_api.py

```diff
@@ -22,21 +23,32 @@ def test_persist_local(local_storage):
 
 @parametrize_all_dataset_storages
 def test_populate_dataset(ds):
-    assert read_dataset_meta(ds.provider) == {"tensors": []}
-    ds.image = np.ones((4, 28, 28))
-    assert read_dataset_meta(ds.provider) == {"tensors": ["image"]}
+    assert ds.meta == {"tensors": []}
```

Comment:
I know not part of this PR, but can you add some "append" tests as well inside a "for loop".
ds = Dataset()
ds.create_tensor("image")
for i in range(0,10):
   ds.image.append(np.ones((4, 28, 28)))